### DISCUSSION TITLE
Auto center alignment of the nickname

### DISCUSSION THREAD

**Discussion by sadness-mugetsu** [SixLabors/ImageSharp.Drawing#85] [Q&A] [answered]
I made a command for the bot to welcome a new user who just came to the server in the form of such a message, but all users have different nickname lengths and I encountered this problem if the user has a short nickname, then it is not beautiful in the center of the screen, how to fix this problem sorry for the poor quality I just need library discord.net
![image](https://user-images.githubusercontent.com/68119409/92623074-dba26e80-f2d6-11ea-88b9-059a8c52b877.png)

**Comment by luoxgwb** [comment]
``` cs
using var image = new Image<Rgba32>(400, 200);

var userName = "Test test";

var userNameTextWidth = (int)TextMeasurer.Measure(userName, new RendererOptions(font));

var userNameStartPointX = image.Width / 2 - userNameTextWidth / 2;

image.Mutate(x => x.DrawText(userName, font, Color.White, new PointF(userNameStartPointX, 100)));
```

**Reply by tocsoft** [reply]
this will cause 2 layout executions, one for the measure and again for the draw, you want to use the `TextGraphicOptions` overload and specify a `HorizontalAlignment`.

**Reply by luoxgwb** [reply]
> this will cause 2 layout executions, one for the measure and again for the draw, you want to use the `TextGraphicOptions` overload and specify a `HorizontalAlignment`.

Thank you for pointing out the problem, it seems that I still need to continue learning

**Reply by tocsoft** [reply]
No worries. By the way, there is nothing actually wrong with your solution and if we didn't have a horizontal alignment setting your solution would have been fine and probably what I would have suggested, we just happen to have a more efficient way of doing it in box.

**Comment by tocsoft** [comment, marked as answer]
You can configure the engine with various alignments, the bellow uses the `Center` alignment with a drawing point of the center of the image this will cause the text to draw outwards left and right evenly from the point specified.

```c#
Image image = ...; // create any way you like.
Font font = ...; // see our Fonts library for best practices on retriving one of these.

TextGraphicsOptions options = new TextGraphicsOptions()
{
    HorizontalAlignment = HorizontalAlignment.Center // center align
};

IBrush brush = Brushes.Horizontal(Color.Red, Color.Blue);
IPen pen = Pens.DashDot(Color.Green, 5);
string text = "sample text";
int halfImageWidth = image.Width / 2;

// draws a star with Horizontal red and blue hatching with a dash dot pattern outline.
image.Mutate( x=> x.DrawText(options, text, font, brush, pen, new PointF(halfImageWidth, 100));
```
